rpc/gnmi/gnmignoissh-dialout-grpctunnel.md

```diff
@@ -27,48 +27,55 @@ Carl Lebsack
 
 ![](img/tunnelclienttotunnelserver.png)
 
 ### Tunnel client to tunnel client session establishment
-This final case is a special case of the Tunnel Client to Tunnel server session establishment whereby the ultimate connection may be routed to another Tunnel Client.  It illustrates the full power of the gRPC tunnel in that a server can act as a midpoint for connections between clients.  A client can, in addition to or instead of registering its available targets, request to be notified of targets reachable by the server.  This special case focuses on the ability to route a single session be relayed by concatenating two independent client sessions within the tunnel server.
+This final case is a special case of the tunnel client to tunnel server session establishment whereby the ultimate connection may be routed to another tunnel client.  It illustrates the full power of the gRPC tunnel in that a tunnel server can act as a midpoint for connections between tunnel clients.  A tunnel client can, in addition to or instead of registering its available targets, request to be notified of targets reachable by the tunnel server.  This special case focuses on the ability to route a single session be relayed by concatenating two independent tunnel client sessions within the tunnel server.
 
 ![](img/tunnelclienttotunnelclient.png)
 
 ## Target subscription
-A client has the option of getting a subscription of all targets registered with the tunnel.  After the Tunnel is established, a client sends a Subscribe message (optionally specifying a target type) and receives a stream of Target messages from the server for all currently registered targets followed by a Subscribe accept.  Future target additions that match the subscription type will be forwarded to the client as Target messages at the time they are added.
+A tunnel client has the option of getting a subscription of all targets registered with the tunnel.  After the Tunnel is established, a tunnel client sends a Subscribe message (optionally specifying a target type) and receives a stream of Target messages from the tunnel server for all currently registered targets followed by a Subscribe accept.  Future target additions that match the subscription type will be forwarded to the tunnel client as Target messages at the time they are added. This can be useful when the tunnel client is acting as a gNMI target in the Tunnel client to tunnel client session establishment case, so it can discovery available relays.
```

Comment:
Replace last sentence.

A subscription is used by a client of the selected tunnelled protocol (e.g. an ssh client for ssh or a gNMI telemetry collector for a gNMI.Subscribe) to discover the list of targets reachable via a given tunnel server.

Reply:
done